src/main/java/org/jabref/gui/groups/GroupTreeView.java

```diff
@@ -199,9 +202,17 @@ private void initialize() {
                                 group.allSelectedEntriesMatchedProperty());
                     }
                     Text text = new Text();
-                    if (preferencesService.getDisplayGroupCount()) {
-                        text.textProperty().bind(group.getHits().asString());
-                    }
+                    EasyBind.subscribe(preferencesService.getGroupsPreferences().displayGroupCountProperty(),
+                            (newValue) -> {
+                                if (text.textProperty().isBound()) {
```

Comment:
Why is this necessary? Please add a comment

Reply:
I cannot set the text to an empty value if the value is bound, instead it keeps the last calculated number of entries in `group.getHits()` or just `0`, if the preference option to calculate the number of entries (which is very expensive in large databases) is deactivated. If the calculation of entries is disabled and you add or remove an entry from a group it would display the wrong number of entries.